mobile-save-for-later/src/main/scala/com/gu/sfl/identity/IdentityService.scala

```diff
@@ -71,4 +86,25 @@ class IdentityServiceImpl(identityConfig: IdentityConfig, okHttpClient: OkHttpCl
     })
     promise.future
   }
+
+  def userFromRequestOauth(identityHeaders: IdentityHeader, requiredScope: List[IdentityAccessScope]): Future[Option[String]] = {
+    val promise = Promise[Option[String]]
+    val claims = oktaLocalValidator.claimsFromAccessToken(identityHeaders.auth.stripPrefix("Bearer "), requiredScope)
+    println(claims.toString)
+    claims match {
+      case Left(e) => {
+        promise.failure(OktaValidationException(e))
+      }
+      case Right(c) => promise.success(Option(c.identityId))
+    }
+
+    promise.future
+  }
+
+  override def userFromRequest(identityHeaders: IdentityHeader, requiredScope: List[IdentityAccessScope]): Future[Option[String]] = {
+    identityHeaders.isOauth match {
+      case true => userFromRequestOauth(identityHeaders, requiredScope)
```

Comment:
If `userFromRequestOauth` is changed so that it doesn't return a `Future`, this will need to become:
```suggestion
      case true => Future.successful(userFromRequestOauth(identityHeaders, requiredScope))
```